packages/create-webstone-app/src/tasks/2-setup-project/plugin.ts

```diff
@@ -0,0 +1,98 @@
+//@ts-ignore this package doesn't provide a declaration file
+import { create } from "create-svelte";
+import { Ctx, getAppName } from "../../helpers";
+import path, { dirname } from "path";
+import fs from "fs-extra";
+import { fileURLToPath } from "url";
+import { ListrTask } from "listr2";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+export const copyTemplate = (ctx: Ctx) => {
+  const appDir = ctx.appDir;
+  fs.copySync(
+    path.join(__dirname, "..", "templates", "plugin", "structure"),
+    appDir
+  );
+};
+
+export const renameCliPackage = (ctx: Ctx) => {
+  const cliPackageJson = fs.readJSONSync(
+    path.join(ctx.appDir, "packages", "cli", "package.json")
+  );
+  cliPackageJson.name = `webstone-plugin-cli-${getAppName(ctx.appDir)}`;
+  fs.writeJSONSync(
+    path.join(ctx.appDir, "packages", "cli", "package.json"),
+    cliPackageJson,
+    {
+      spaces: "\t",
+    }
+  );
+};
+
+export const renameMainPackage = (ctx: Ctx) => {
+  const packageJson = fs.readJSONSync(path.join(ctx.appDir, "package.json"));
+  packageJson.name = `webstone-plugin-${getAppName(ctx.appDir)}`;
+  fs.writeJSONSync(path.join(ctx.appDir, "package.json"), packageJson, {
+    spaces: "\t",
+  });
+};
+
+export const setPrivateTrue = (ctx: Ctx) => {
+  const packageJson = fs.readJSONSync(path.join(ctx.appDir, "package.json"));
+  packageJson.private = true;
+  fs.writeJSONSync(path.join(ctx.appDir, "package.json"), packageJson, {
+    spaces: "\t",
+  });
+};
+
+export const renamePackages = async (ctx: Ctx) => {
```

Comment:
We may need to update the function name, when doing other stuff than just changing the name of the package

Reply:
Big mess up, fix incoming :)